extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -859,18 +867,19 @@ export class AmpA4A extends AMP.BaseElement {
       }
       if (!creativeMetaData) {
         // Non-AMP creative case, will verify ad url existence.
-        return this.renderNonAmpCreative_();
+        return this.renderNonAmpCreative_(checkStillCurrent);
```

Comment:
I don't think you actually need to pass checkStillCurrent around if you believe you have checked the current promiseId against the global prior to any promise releases.  You should just be able to get the current promiseId prior to promise and compare.